packages/frontend/src/pages/ParkingSpaces/components/CloseListing.tsx

```diff
@@ -0,0 +1,112 @@
+import { useState } from 'react'
+import {
+  Button,
+  Dialog,
+  DialogTitle,
+  DialogContent,
+  Typography,
+  Box,
+  IconButton,
+} from '@mui/material'
+import CloseIcon from '@mui/icons-material/Close'
+import { toast } from 'react-toastify'
+import { LoadingButton } from '@mui/lab'
+
+import { useCloseParkingSpaceListing } from '../hooks/useCloseParkingSpaceListing'
+
+export const CloseListing = (props: { listingId: number }) => {
+  const closeListing = useCloseParkingSpaceListing()
+  const [open, setOpen] = useState(false)
+
+  const onClose = () => {
+    setOpen(false)
+    closeListing.reset()
+  }
+
+  const onCloseListing = () =>
+    closeListing.mutate(
+      { listingId: props.listingId },
+      {
+        onSuccess: () => {
+          setOpen(false)
+          toast('Bilplatsannons markerad som ompublicerad', {
+            type: 'success',
+            hideProgressBar: true,
+          })
+        },
+      }
+    )
+
+  return (
+    <>
+      <Button variant="dark" onClick={() => setOpen(true)}>
+        Markera som publicerad
+      </Button>
+      <Dialog
+        onClose={onClose}
+        open={open}
+        maxWidth="xs"
+        TransitionProps={{ exit: false }}
+      >
+        <Box paddingTop="0.5rem">
```

Comment:
This component looks quite similar to the `ActionDialog` component? We might be able to use that here instead of writing new markup?

Reply:
[bfbc99d](https://github.com/Bostads-AB-Mimer/onecore-internal-portal/pull/57/commits/bfbc99d27847e64723d1acefcd545e22c7020c16)

[ab65bff](https://github.com/Bostads-AB-Mimer/onecore-internal-portal/pull/57/commits/ab65bff6addae6156b5da1f432f3ad2777b11db0)